src/main/java/org/broadinstitute/hellbender/tools/exome/SegmentUtils.java

```diff
@@ -213,6 +217,30 @@ public static SimpleInterval trimInterval(final Locatable interval,
                 dataLine -> toModeledSegment(dataLine, false));
     }
 
+    /** Read only the sample names from a segment file.
+     *
+     * @param segmentFile Never {@code null}
+     * @return a list of the unique sample names in the seg file.
+     */
+    public static List<String> readSampleNamesFromSegmentFile(final File segmentFile) {
```

Comment:
This should use a one-liner to read in the list of all names:
`readSegmentFile(segmentFile, new String[]{SegmentTableColumns.SAMPLE.toString()}, SegmentUtils::toSampleName)`
You can then convert this list to a Set of unique names and back to a list, if desired.

When I previously refactored this class, I extracted all of the boilerplate TableReader code, checking of mandatory columns, and exception throwing to the private readSegmentFile method.  This got rid of a lot of copy/pasted code that was common to each readXFile method (where X = Segment, ModeledSegment, etc.)---all of these methods could be turned into one-liners that simply pass the mandatory columns and a lambda for translating the dataLine to X.  Same goes for the writeSegmentFile method.

However, I see that @vruano is organizing IO of germline segments in a different manner.  So we should definitely standardize when refactoring all the segment stuff.  @davidbenjamin